use-cases/using-SPDK-vhostuser-and-kata.md

```diff
@@ -0,0 +1,221 @@
+|      |                                |
++----------------------------+--+-------+
+|      +---------------------+ kvm.ko   |
+|                            +--+-------+
+|                kernel                 |
++---------------------------------------+
+```
+
+SPDK vhost is a vhost-user slave server. It exposes Unix domain sockets and
+allows external applications to connect. It is capable of exposing virtualized
+storage devices to QEMU instances or other arbitrary processes.
+
+Currently, the SPDK vhost-user target can exposes these types of virtualized
+devices:
+
+- `vhost-user-blk`
+- `vhost-user-scsi`
+- `vhost-user-nvme`
+
+For more information, visit [SPDK](https://spdk.io) and [SPDK vhost-user target](https://spdk.io/doc/vhost.html).
+
+## Install and setup SPDK vhost-user target
+
+### Get source code and build SPDK
+
+Following the SPDK [getting started guide](https://spdk.io/doc/getting_started.html).
+
+### Run SPDK vhost-user target
+
+First, run the SPDK `setup.sh` script to setup some hugepages for the SPDK vhost
+target application. We recommend you use a minimum of 4GiB, enough for the SPDK
+vhost target and the virtual machine.
+This will allocate 4096MiB (4GiB) of hugepages:
+
+```bash
+$ sudo HUGEMEM=4096 scripts/setup.sh
+```
+
+Then, take directory `/var/run/kata-containers/vhost-user` as Kata's vhost-user
+device directory. Make subdirectories for vhost-user sockets and device nodes:
+
+```bash
+$ sudo mkdir -p /var/run/kata-containers/vhost-user/
+$ sudo mkdir -p /var/run/kata-containers/vhost-user/block/
+$ sudo mkdir -p /var/run/kata-containers/vhost-user/block/sockets/
+$ sudo mkdir -p /var/run/kata-containers/vhost-user/block/devices/
+```
+
+For more details, see section [Host setup for vhost-user devices](#host-setup-for-vhost-user-devices).
+
+Next, start the SPDK vhost target application.  The following command will start
+vhost on the first CPU core with all future socket files placed in
+`/var/run/kata-containers/vhost-user/block/sockets/`:
+
+```bash
+$ sudo app/spdk_tgt/spdk_tgt -S /var/run/kata-containers/vhost-user/block/sockets/ &
+```
+
+To list all available vhost options run the following command:
+
+```bash
+$ app/spdk_tgt/spdk_tgt -h
+```
+
+Create an experimental `vhost-user-blk` device based on memory directly:
+
+- The following RPC will create a 64MB memory block device named `Malloc0`
+with 4096-byte block size:
+
+```bash
+$ sudo scripts/rpc.py bdev_malloc_create 64 4096 -b Malloc0
+```
+
+- The following RPC will create a `vhost-user-blk` device exposing `Malloc0`
+block device. The device will be accessible via
+`/var/run/kata-containers/vhost-user/block/sockets/vhostblk0`:
+
+```bash
+$ sudo scripts/rpc.py vhost_create_blk_controller vhostblk0 Malloc0
+```
+
+## Host setup for vhost-user devices
+
+Considering the OCI specification and characteristics of vhost-user device,
+Linux reserves the block major range `240-254` to map each vhost-user block
```

Comment:
These ranges are not really for vhost-user specifically iiuc, but for local use. We with Kata have chosen to use these for vhost-user devices. If that is correct, we should change the wording to reflect this instead.